File: journal-monitor/internal/model/article.go
package model

import "time"

type Article struct {
	ID          string    `json:"id"`
	DOI         string    `json:"doi"`
	Title       string    `json:"title"`
	Authors     []string  `json:"authors"`
	Abstract    string    `json:"abstract"`
	JournalID   string    `json:"journal_id"`
	PublishDate *time.Time `json:"publish_date,omitempty"`
	URL         string    `json:"url"`
	FetchedAt   time.Time `json:"fetched_at"`
}

File: app/models/article.py
import uuid
from datetime import date, datetime

from sqlalchemy import Date, DateTime, ForeignKey, String, Text, func
from sqlalchemy.dialects.postgresql import ARRAY, UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.db import Base


class Article(Base):
    __tablename__ = "articles"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid()
    )
    doi: Mapped[str | None] = mapped_column(String(255), nullable=True, server_default=None)
    title: Mapped[str] = mapped_column(Text, nullable=False)
    authors: Mapped[list[str]] = mapped_column(
        ARRAY(Text), nullable=False, server_default="{}"
    )
    abstract: Mapped[str] = mapped_column(Text, nullable=False, server_default="")
    journal_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("journals.id", ondelete="CASCADE"), nullable=False
    )
    publish_date: Mapped[date | None] = mapped_column(Date, nullable=True)
    url: Mapped[str] = mapped_column(Text, nullable=False, server_default="")
    fetched_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )
